### ISSUE TITLE
observation : Inconsistent NoLivingConnectionsError while regarding a new application

### ISSUE THREAD

**Issue by PrathibaJee** [openBackhaul/ApplicationLayerTopology#544]
Noticed the following error inconsistently while regarding an application to the ApplicationLayerTopology.
```
NoLivingConnectionsError: Given the configuration, the ConnectionPool was not able to find a usable Connection for this request.
    at makeRequest (C:\Users\NQ10024811\Desktop\SDNApplicationLayerVersion2.1.1\ApplicationLayerTopology\server\node_modules\@elastic\elasticsearch\lib\Transport.js:203:43)
    at prepareRequest (C:\Users\NQ10024811\Desktop\SDNApplicationLayerVersion2.1.1\ApplicationLayerTopology\server\node_modules\@elastic\elasticsearch\lib\Transport.js:460:11)
    at Transport.request (C:\Users\NQ10024811\Desktop\SDNApplicationLayerVersion2.1.1\ApplicationLayerTopology\server\node_modules\@elastic\elasticsearch\lib\Transport.js:503:7)
    at Client.searchApi [as search] (C:\Users\NQ10024811\Desktop\SDNApplicationLayerVersion2.1.1\ApplicationLayerTopology\server\node_modules\@elastic\elasticsearch\api\api\search.js:61:25)
    at getLinkOfTheOperationAsync (C:\Users\NQ10024811\Desktop\SDNApplicationLayerVersion2.1.1\ApplicationLayerTopology\server\service\individualServices\LinkServices.js:206:28)
    at async exports.prepareLinkChangeNotificationForwardingsAsync (C:\Users\NQ10024811\Desktop\SDNApplicationLayerVersion2.1.1\ApplicationLayerTopology\server\service\individualServices\LinkServices.js:568:24)
    at async UpdateControlConstructAndLinksInDataBase (C:\Users\NQ10024811\Desktop\SDNApplicationLayerVersion2.1.1\ApplicationLayerTopology\server\service\individualServices\regardApplicationAutomation.js:507:24)
    at async RequestForInquiringTopologyChangeInformationWithDefaultKey (C:\Users\NQ10024811\Desktop\SDNApplicationLayerVersion2.1.1\ApplicationLayerTopology\server\service\individualServices\regardApplicationAutomation.js:115:39)
    at async exports.regardApplication (C:\Users\NQ10024811\Desktop\SDNApplicationLayerVersion2.1.1\ApplicationLayerTopology\server\service\individualServices\regardApplicationAutomation.js:31:14)
    at async exports.regardApplication (C:\Users\NQ10024811\Desktop\SDNApplicationLayerVersion2.1.1\ApplicationLayerTopology\server\service\IndividualServicesService.js:767:20) {
  meta: undefined
}
NewApplicationCausesRequestForTopologyChangeInformation.RequestForInquiringTopologyChangeInformationWithDefaultKey has been triggered with response 200
```
Issue occurring inconsistently , let me know whether you too observe this issue.

**Comment by IswaryaaS** [comment]
we could see, it occurs inconsistently and under different places. 
As discussed, we can include a retry mechanism while we are getting this error on connection to ES.

**Comment by PrathibaJee** [comment]
@IswaryaaS , many thanks for helping in resolving this issue.
Issue merged in PR #546 , hence closing this.
